web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -192,6 +193,10 @@
   }
 
   onMount(async () => {
+    if (isSharedLink()) {
+      sharedLink = await getMySharedLink({ key: getKey() });
```

Comment:
Will this load the shared link every time the component is opened? This might not be ideal. Another option would be to use the context api or a store for tracking state related to the shared link.